### PR TITLE
migration and create model, relationship

### PR DESCRIPTION
**Pull request by hngda01**
image

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -59,3 +59,7 @@ config/environments/*.local.yml
 /.idea/*
 
 /config/application.yml
+
+config/settings.local.yml
```
Comment (tuquyet2709) [inline .gitignore]: xoá 3 dòng này, trùng rồi

File: app/models/group.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class Group < ApplicationRecord
+  has_many :group_members
+  has_many :members, through: :group_members, source: :user
+  has_many :tasks
+  belongs_to :leader, class_name: "User", foreign_key: "leader_id"
+  validates :leader_id, presence: true
+
```
Comment (tuquyet2709) [inline app/models/group.rb]: xoá dòng

**Comment by tuquyet2709** [comment]
ok

File: app/models/relationship.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+class Relationship < ApplicationRecord
+  belongs_to :follower, class_name: "User"
```
Comment (thanhhungchu95) [inline app/models/relationship.rb]: class_name: User.name

File: app/models/user.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
+class User < ApplicationRecord
+  has_many :lead_groups, class_name: "Group", foreign_key: "leader_id"
```
Comment (thanhhungchu95) [inline app/models/user.rb]: class_name: Group.name

File: app/models/user.rb [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  has_many :group_members, foreign_key: "member_id"
+  has_many :groups, through: :group_members, source: :group
+  has_many :reports, foreign_key: "member_id"
+  has_many :active_relationships, class_name: "Relationship",
+    foreign_key: "follower_id",
+    dependent: :destroy
+  has_many :following, through: :active_relationships, source: :followed
+  has_many :passive_relationships, class_name: "Relationship",
```
Comment (thanhhungchu95) [inline app/models/user.rb]: class_name: Relationship.name

File: app/models/user.rb [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    uniqueness: {case_sensitive: false}
+  has_secure_password
+  validates :password, presence: true,
+    length: {minimum: Settings.minimum.length_pass},
+    allow_nil: true
+
+  private
+
+  def downcase_email
+    self.email = email.downcase
```
Comment (thanhhungchu95) [inline app/models/user.rb]: downcase!

**Comment by tuquyet2709** [comment]
ok

**Pull request merged**